### ISSUE TITLE
Feature suggestion: `.explore()` method for geometry objects and bounding boxes

### ISSUE THREAD

**Issue by robbibt** [opendatacube/odc-geo#125]
After sharing the new `.odc.explore()` method within DEA, it was suggested that a similar `.explore()`-style method would be a valuable addition to other `odc-geo` geometry objects too. 

In particular, having an `.explore()` style method available on `Geometry` and `BoundingBox` objects (or even potentially `GeoBoxes` via their extent?) that plots these objects on an interactive map would allow users to:
- Easily verify that the extents/shape of their geometries line up with their expected AOI
- Bugfix issues when creating complex Geometries
- Visually communicate to beginner users where they are about to load or analyse data

It would also add consistency with other geometry-related packages like `GeoPandas` which also provide an `.explore()` method for their spatial data classes.

**Comment by Kirill888** [comment]
I agree that this would be handy, but in my experience I always want to display a combination of different geoboxes, geometries, bounding boxes, and I want syntax for that to be easy to type in the notebook, while still allowing for fine-tuning presentations of individual components. 

On the base level we can start with `explore(..., map=None|folium.Map|ipyleaflet.Map)`, but at user level I'd like a shorter and more "left to right" interface like:

```python
m = folium.Map(....)
gbox1 @ {'color': 'green'} >> gbox2 >> xx.odc.thumbnail(robust=True) >> m
``` 

where `>> m` can be omitted.

Honestly, lack of that functionality is a significant contributor to the absence of "pixel grid documentation", as it's just too hard to visualize all the needed concepts with "raw" matplotlib or SVG.

**Comment by robbibt** [comment]
> I agree that this would be handy, but in my experience I always want to display a combination of different geoboxes, geometries, bounding boxes, and I want syntax for that to be easy to type in the notebook, while still allowing for fine-tuning presentations of individual components.
> 
> On the base level we can start with `explore(..., map=None|folium.Map|ipyleaflet.Map)`, but at user level I'd like a shorter and more "left to right" interface like:
> 
> ```python
> m = folium.Map(....)
> gbox1 @ {'color': 'green'} >> gbox2 >> xx.odc.thumbnail(robust=True) >> m
> ```
> 
> where `>> m` can be omitted.
> 
> Honestly, lack of that functionality is a significant contributor to the absence of "pixel grid documentation", as it's just too hard to visualize all the needed concepts with "raw" matplotlib or SVG.

That would be really neat - reminds me of `%>%` piping and chaining different plotting functions together in R's `ggplot2`. 

My feeling is that these aren't mutually exclusive - `.explore()` serves the purpose of easily plotting very simple interactive maps to quickly explore a dataset at a glance, while the proposed syntax above would be perfect for producing highly customised and fine tuned plots.

**Comment by robbibt** [comment]
Closed by #128
